core/src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java

```diff
@@ -429,4 +430,50 @@ private Component getChatFormat(@Nullable Component prefix, MatchPlayer player,
         .append(msg)
         .build();
   }
+
+  private void trackMessage(Player receiver, Player sender) {
+    MessageSenderIdentity senderIdent = new MessageSenderIdentity(receiver, sender);
+    this.lastMessagedBy.put(receiver, senderIdent);
+  }
+
+  private UUID getLastMessagedId(Player sender) {
+    MessageSenderIdentity targetIdent = lastMessagedBy.get(sender);
+    if (targetIdent == null) return null;
+    MatchPlayer target = manager.getPlayer(targetIdent.getPlayerId());
+
+    // Prevent replying to offline players
+    if (target == null) return null;
+
+    // Compare last known and current name
+    String lastKnownName = targetIdent.getName();
+    String currentName = Players.getVisibleName(sender, target.getBukkit());
+
+    boolean reveal = Players.shouldReveal(sender, target.getBukkit());
```

Comment:
i believe reveal here shouldn't be needed, because shouldReveal is a an and of isVisible (checking for vanish) + getVisibleName (checking for if can see the nick), which you're essentially reimplementing here yourself